package/MDAnalysis/units.py

```diff
@@ -343,16 +343,14 @@ def convert(x, u1, u2):
     try:
         ut1 = unit_types[u1]
     except KeyError:
-        raise ValueError(("unit '{0}' not recognized.\n" +
-                          len("ValueError: ")*[" "] +
-                          "It must be one of {1}.").format(u1, unit_types))
+        raise ValueError("unit '{0}' not recognized.\n"
+                          "It must be one of {1}.".format(u1, ", ".join(unit_types)))
```

Comment:
This breaks on python 3 with the following exception:

```python
Traceback (most recent call last):
  File "/home/jon/Envs/mda3/lib/python3.4/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/jon/dev/mda3/testsuite/MDAnalysisTests/test_units.py", line 109, in test_unit_unknown
    assert_raises(ValueError, units.convert, nm, 'Stone', 'nm')
  File "/home/jon/Envs/mda3/lib/python3.4/site-packages/numpy/testing/utils.py", line 1109, in assert_raises
    return nose.tools.assert_raises(*args,**kwargs)
  File "/usr/lib/python3.4/unittest/case.py", line 704, in assertRaises
    return context.handle('assertRaises', callableObj, args, kwargs)
  File "/usr/lib/python3.4/unittest/case.py", line 162, in handle
    callable_obj(*args, **kwargs)
  File "/home/jon/dev/mda3/package/MDAnalysis/units.py", line 347, in convert
    "It must be one of {1}.".format(u1, ", ".join(unit_types)))
TypeError: sequence item 38: expected str instance, bytes found
```

This is because we have the symbol for Angstrom as bytes and as unicode. This was done in purpose, but does anybody uses the utf-8 escape code for 	Å (`b'\xe2\x84\xab'`)? Do we really need to support it while we already support `'Angtrom'`, `'A'`, `'angstrom`', and `u'\u212b'`, and that we support `b'\xe2\x84\xab'` only for speed, force, and density units, but not length?

cc @MDAnalysis/coredevs